test/demo/utils/fdc3RemoteExecution.ts

```diff
@@ -43,25 +43,87 @@ export async function broadcast(executionTarget: Identity, context: Context): Pr
         .then(() => new Promise<void>(res => setTimeout(res, 100)));  // Broadcast is fire-and-forget. Slight delay to allow for service to handle
 }
 
-/**
- * This will register a simple contextListener on the window.
- *
- * Returns a promise which resolves to the context passed to the listener
- */
-export async function addContextListener(executionTarget: Identity, handler: (context: Context) => void): Promise<void> {
-    return ofBrowser
-        .executeOnWindow(
-            executionTarget,
-            async function(this: TestWindowContext):
-                Promise<Context> {
-                return new Promise<Context>(res => {
-                    const listener = this.OpenfinFDC3.addContextListener((context: Context) => {
-                        res(context);
-                    });
-                });
-            }
-        )
-        .then(async (result) => {
-            handler(await result);
+export async function getAllChannels(executionTarget: Identity): Promise<Channel[]> {
+    return ofBrowser.executeOnWindow(executionTarget, function(this: TestWindowContext): Promise<Channel[]> {
+        return this.OpenfinFDC3.getAllChannels();
+    });
+}
+
+export async function joinChannel(executionTarget: Identity, channelId: ChannelId, identity?: Identity): Promise<void> {
+    return ofBrowser.executeOnWindow(executionTarget, function(this: TestWindowContext, channelId: ChannelId, identity?: Identity): Promise<void> {
+        return this.OpenfinFDC3.joinChannel(channelId, identity);
+    }, channelId, identity);
+}
+
+export async function getChannel(executionTarget: Identity, identity?: Identity): Promise<Channel> {
+    return ofBrowser.executeOnWindow(executionTarget, function(this: TestWindowContext, identity?: Identity): Promise<Channel> {
+        return this.OpenfinFDC3.getChannel(identity);
+    }, identity);
+}
+
+export async function getChannelMembers(executionTarget: Identity, channelId: ChannelId): Promise<Identity[]> {
+    return ofBrowser.executeOnWindow(executionTarget, function(this: TestWindowContext, channelId: ChannelId): Promise<Identity[]> {
+        return this.OpenfinFDC3.getChannelMembers(channelId);
+    }, channelId);
+}
+
+export interface RemoteContextListener {
+    type: 'context';
+    remoteIdentity: Identity;
+    id: number;
+    unsubscribe: () => Promise<void>;
+    getReceivedContexts: () => Promise<Context[]>;
+}
+
+export async function addContextListener(executionTarget: Identity): Promise<RemoteContextListener> {
+    const id = await ofBrowser.executeOnWindow(executionTarget, function(this:TestWindowContext): number {
+        const listenerID = this.contextListeners.length;
+        this.contextListeners[listenerID] = this.OpenfinFDC3.addContextListener((context) => {
+            this.receivedContexts.push({listenerID, context});
         });
+        return listenerID;
+    });
+
+    return {
+        type: 'context',
```

Comment:
Yea, the utils as they currently stand don't include any stuff for intent listeners. This was in place to allow a more unified "mountEventListener" util which is used by addContextListener and addIntentListener once I got the intent listener stuff in. Since I'm no longer including that in this story I could probably remove it.